src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/InterchangeHeader.java

```diff
@@ -34,26 +41,29 @@ public String getKey() {
     public String getValue() {
         String timestamp = DATE_FORMAT.format(translationTime);
         String formattedSequenceNumber = String.format("%08d", sequenceNumber);
-        return "UNOA:2"+"+"+sender+"+"+recipient+"+"+timestamp+"+"+formattedSequenceNumber;
+
+        return List.of(UNOA_PREFIX, sender, recipient, timestamp, formattedSequenceNumber, transferString).stream()
+                .filter(StringUtils::isNotBlank)
```

Comment:
Is "+" concatenation not simpler?

What happens if a field **is** blank? This would not be a valid segment both because its missing data and because the number of delimiters is wrong.

Reply:
refacored